src/index.js

```diff
@@ -1,13 +1,41 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import MovieDetails from "./components/MovieDetails";
+import ShowMoviesSearch from "./components/ShowMoviesSearch";
+import TvShowDetails from "./components/TvShowDetails";
+import PersonDetails from "./components/PersonDetails";
 
-const root = ReactDOM.createRoot(document.getElementById('root'));
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+  },
+  {
+    path: "results",
+    element: <ShowMoviesSearch />,
+  },
+  {
+    path: "movie/:id",
+    element: <MovieDetails />,
+  },
+  {
+    path: "tvshow/:id",
+    element: <TvShowDetails />,
+  },
+  {
+    path: "person/:id",
+    element: <PersonDetails />,
+  },
+]);
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
   <React.StrictMode>
-    <App />
+    <RouterProvider router={router} />
```

Comment:
you can put the <Header /> above this, so that you don't call it in every component :)